runtime/codegen/encoder.go

```diff
@@ -329,14 +329,21 @@ func (e *Encoder) Interface(value any) {
 		panic(fmt.Sprintf("cannot encode unknown type %T since it is not marked as AutomMarshal", value))
 	}
 
-	// We want a pointer to the value wrapped in any.
-	// XXX Support pointers to structs as well.
-	copy := reflect.New(t)
-	copy.Elem().Set(reflect.ValueOf(value))
-	am, ok := copy.Interface().(AutoMarshal)
+	// We allow either value, or a pointer to value to implement AutoMarshal.
+	am, ok := value.(AutoMarshal)
+	if !ok {
+		// Try using a pointer to value.
```

Comment:
Can we use the trick you used in `automarshal.go` here? 

```go
if !ok {
    value = &value
    am, ok = value.(AutoMarshal)
}
```

Reply:
That was a good idea. Until I realized that the "a = &value" trick was only working because value had the generic type T. Here, value has type any, and so "&value" does not have the type we want.